torch/distributed/checkpoint/state_dict_loader.py

```diff
@@ -180,7 +180,8 @@ def load(
             elem = state_dict[key]
             if isinstance(elem, Stateful):
                 elem.load_state_dict(statetful_sd[key])
-            state_dict[key] = statetful_sd[key]
+            else:
+                state_dict[key] = statetful_sd[key]
```

Comment:
Not related to this PR, but I wonder if we could add more comments to the codes from line 161 to line 184. Much appreciated!
Cc: @fegin @LucasLLC